mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -147,4 +157,47 @@ private boolean belongsToSameUser(Mailbox mailbox, Mailbox otherMailbox) {
     public void updateACL(Mailbox mailbox, MailboxACL.MailboxACLCommand mailboxACLCommand) throws MailboxException{
         mailbox.setACL(mailbox.getACL().apply(mailboxACLCommand));
     }
+
+    @Override
+    public List<MailboxAnnotation> getAnnotations(Mailbox mailbox) throws MailboxException {
+        InMemoryId maiboxId = (InMemoryId) mailbox.getMailboxId();
+        return Lists.newArrayList(retrieveAllAnnotations(maiboxId));
+    }
+
+    private Iterable<MailboxAnnotation> retrieveAllAnnotations(InMemoryId maiboxId) {
+        return Iterables.transform(
```

Comment:
I really don't like locking on an object, please use a real Lock object (or anything that would fit)